web/src/message_fetch.ts

```diff
@@ -715,6 +715,39 @@ export function initialize(finished_initial_fetch: () => void): void {
         } catch (error) {
             blueslip.error("Error in recent_view_ui.process_messages", undefined, error);
         }
+
+        // Since we backfill a lot more messages here compared to rendered message list,
+        // we can try populating them if we can do so locally.
+        for (const msg_list of message_lists.all_rendered_message_lists()) {
+            if (!msg_list.data.filter.can_apply_locally()) {
+                continue;
+            }
+
+            if (msg_list.empty() && msg_list.data.fetch_status.is_loading_messages()) {
+                // This is the first message fetch for this message list, we don't
+                // want to add messages right now, since we don't have a range of
+                // messages that user wants to see.
+                continue;
+            }
```

Comment:
This comment isn't correct; `is_loading_messages` is true whenever there's an active request to the server to fetch messages, regardless of if it's the first one.

Reply:
I removed this condition since we already skip adding messages on `msg_lsit.visibily_empty()` (which I added later to this commit and didn't realize that this block become redundant).